app/views/layouts/application.html.erb

```diff
@@ -1,5 +1,13 @@
 <%= render "shell/top" %>
-<style>.app-shell-loader {display: none;}</style>
+<% if internal_navigation? %>
+  <style>.app-shell-loader {display: none;}</style>
+  <script async>
```

Comment:
`async` because this script doesn't need to interrupt rendering, but not `defer` because we don't need to weight for other html or scripts.